python/ray/serve/_private/deployment_scheduler.py

```diff
@@ -519,13 +527,27 @@ def _schedule_replica(
                 if scheduling_request.placement_group_strategy
                 else "PACK"
             )
-            pg = self._create_placement_group_fn(
-                scheduling_request.placement_group_bundles,
-                placement_group_strategy,
-                _soft_target_node_id=target_node_id,
-                lifetime="detached",
-                name=scheduling_request.actor_options["name"],
-            )
+            try:
+                pg = self._create_placement_group_fn(
+                    scheduling_request.placement_group_bundles,
+                    placement_group_strategy,
+                    _soft_target_node_id=target_node_id,
+                    lifetime="detached",
+                    name=scheduling_request.actor_options["name"],
+                )
+            except Exception:
+                # We add a defensive exception here, so the controller can
+                # make progress even if the placement group isn't created.
+                # See https://github.com/ray-project/ray/issues/43888.
+                logger.exception(
+                    f"Failed to create a placement group for {replica_id}."
+                )
+                scheduling_request.scheduling_failed = True
+                scheduling_request.scheduling_failed_reason = (
```

Comment:
Hmm, `scheduling_request` is a type `ReplicaSchedulingRequest` right? How does `scheduling_failed` and `scheduling_failed_reason` work?

Reply:
Ok, I see how this is used below. Maybe we can add those new attributes to  `ReplicaSchedulingRequest`? Also, would be nice to make the failed reason some kind of enum and can move replica id to `_handle_scheduling_request_failures()` when constructing the error message to be more extendable. 

PS, can probable just have one single field and have a success code and failed codes, then in `_handle_scheduling_request_failures()` only check if the code is not success.

Reply:
> Maybe we can add those new attributes to `ReplicaSchedulingRequest`

I'm confused– this PR does add those attributes to the `ReplicaSchedulingRequest`. Was there another object you felt we should add them to?

Reply:
I collapsed the fields into a single `status` field that's populated by an enum.